core/src/dird/ua_output.cc

```diff
@@ -619,146 +674,89 @@ static bool DoListCmd(UaContext* ua, const char* cmd, e_list_type llist)
-    ua->ErrorMsg(T_("invalid jobstatus parameter\n"));
-    return false;
-  }
+static bool DoListCmd(UaContext* ua, const char* cmd, e_list_type llist)
+{
+  if (!OpenClientDb(ua, true)) { return true; }
 
-  // joblevel=X
-  std::vector<char> joblevel_list;
-  if (!GetUserJobLevelSelection(ua, joblevel_list)) {
-    ua->ErrorMsg(T_("invalid joblevel parameter\n"));
-    return false;
-  }
+  Dmsg1(20, "list: %s\n", cmd);
 
-  // jobtype=X
-  std::vector<char> jobtypes{};
-  if (!GetUserJobTypeListSelection(ua, jobtypes, false)) {
-    ua->ErrorMsg(T_("invalid jobtype parameter\n"));
+  if (ua->argc <= 1) {
+    ua->ErrorMsg(T_("%s command requires a keyword\n"), NPRT(ua->argk[0]));
     return false;
   }
 
-  ListCmdOptions optionslist(ua);
+  ListCmdOptions optionslist{};
+  if (!optionslist.parse(ua)) { return false; }
 
   // Select what to do based on the first argument.
-  if ((Bstrcasecmp(ua->argk[1], NT_("jobs")) && (ua->argv[1] == NULL))
-      || ((Bstrcasecmp(ua->argk[1], NT_("job"))
-           || Bstrcasecmp(ua->argk[1], NT_("jobname")))
-          && ua->argv[1])) {
-    // List jobs or List job=xxx
-    if (const char* value; (value = GetArgValue(ua, NT_("jobname")))
-                           || (value = GetArgValue(ua, NT_("job")))) {
-      jr.JobId = 0;
-      bstrncpy(jr.Name, value, MAX_NAME_LENGTH);
-    }
-
-    const char* volumename = GetArgValue(ua, NT_("volume"));
-    const char* poolname = GetArgValue(ua, NT_("pool"));
-
-    switch (llist) {
-      case VERT_LIST:
-        if (!optionslist.count) {  // count result is one column, no filtering
-          SetAclFilter(ua, 2, Job_ACL);
-          SetAclFilter(ua, 7, Client_ACL);
-          SetAclFilter(ua, 22, Pool_ACL);
-          SetAclFilter(ua, 25, FileSet_ACL);
-          if (optionslist.current) {
-            SetResFilter(ua, 2, R_JOB);
-            SetResFilter(ua, 7, R_CLIENT);
-            SetResFilter(ua, 22, R_POOL);
-            SetResFilter(ua, 25, R_FILESET);
-          }
-        }
-        if (optionslist.enabled) { SetEnabledFilter(ua, 2, R_JOB); }
-        if (optionslist.disabled) { SetDisabledFilter(ua, 2, R_JOB); }
-        break;
-      default:
-        if (!optionslist.count) {  // count result is one column, no filtering
-          SetAclFilter(ua, 1, Job_ACL);
-          SetAclFilter(ua, 2, Client_ACL);
-          if (optionslist.current) {
-            SetResFilter(ua, 1, R_JOB);
-            SetResFilter(ua, 2, R_CLIENT);
-          }
-        }
-        if (optionslist.enabled) { SetEnabledFilter(ua, 1, R_JOB); }
-        if (optionslist.disabled) { SetDisabledFilter(ua, 1, R_JOB); }
-        break;
-    }
+  if (Bstrcasecmp(ua->argk[1], NT_("jobs"))
+      || Bstrcasecmp(ua->argk[1], NT_("job"))
+      || Bstrcasecmp(ua->argk[1], NT_("jobname"))
+      || Bstrcasecmp(ua->argk[1], NT_("jobid"))
+      || Bstrcasecmp(ua->argk[1], NT_("ujobid"))) {
+    return ListJobs(ua, llist, optionslist);
+  }
```

Comment:
Im not sure why we moved this from the if-else chain down below.

Reply:
I tried to make the code better readable by this. I now applied the early return also to the rest of the function.